client/core/core.go

```diff
@@ -1373,77 +1374,96 @@ func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) []*DEXBrief {
-					result.AuthErr = details
-					continue
+			authDEXerr := c.authDEX(dc)
+			var mErr *msgjson.Error
+			if authDEXerr != nil {
+				if errors.As(authDEXerr, &mErr) && mErr.Code == msgjson.AccountNotFoundError {
+					acctInfo, err := c.db.Account(dc.acct.id)
+					if err != nil {
+						log.Errorf("Error retrieving account: %v", err)
+					}
+					if !acctInfo.Disabled {
+						details := fmt.Sprintf("%s: %v", dc.acct.host, authDEXerr)
+						c.notify(newFeePaymentNote("DEX auth error", details, db.ErrorLevel, dc.acct.host))
+						result.AuthErr = details
+						err = c.db.DisableAccount(acctInfo)
+						if err != nil {
+							log.Errorf("Error disabling account: %v", err)
+						}
+					}
+					return
 				}
 			}
-			c.reFee(dcrWallet, dc)
-			continue
-		}
-
-		wg.Add(1)
-		go func(dc *dexConnection) {
-			defer wg.Done()
-			err := c.authDEX(dc)
-			if err != nil {
-				details := fmt.Sprintf("%s: %v", dc.acct.host, err)
+			result.Authed = true
+			enabledDEXConnections[host] = dc
+			// The user is authed if the dex account is successfully authenticated against
+			// Now do some house cleaning in regard to registration fee payment
+			if errors.As(authDEXerr, &mErr) && mErr.Code == msgjson.UnpaidAccountError {
+				details := fmt.Sprintf("%s: %v", dc.acct.host, authDEXerr)
 				c.notify(newFeePaymentNote("DEX auth error", details, db.ErrorLevel, dc.acct.host))
-				result.AuthErr = details
+				result.AuthErr += details
+			}
+			if dc.acct.authed() {
 				return
 			}
-			result.Authed = true
-		}(dc)
+			dcrID, _ := dex.BipSymbolID("dcr")
+			if !dc.acct.feePaid() {
+				if len(dc.acct.feeCoin) == 0 {
+					details := fmt.Sprintf("Empty fee coin for %s.", dc.acct.host)
+					c.notify(newFeePaymentNote("Fee coin error", details, db.ErrorLevel, dc.acct.host))
+					result.AuthErr += details
+					return
+				}
+				// Try to unlock the Decred wallet, which should run the reFee cycle, and
+				// in turn will run authDEX.
+				dcrWallet, err := c.connectedWallet(dcrID)
+				if err != nil {
+					log.Debugf("Failed to connect for reFee at %s with error: %v", dc.acct.host, err)
+					details := fmt.Sprintf("Incomplete registration detected for %s, but failed to connect to the Decred wallet", dc.acct.host)
+					c.notify(newFeePaymentNote("Wallet connection warning", details, db.WarningLevel, dc.acct.host))
+					result.AuthErr += details
+					return
+				}
+				if !dcrWallet.unlocked() {
+					err = unlockWallet(dcrWallet, crypter)
+					if err != nil {
+						details := fmt.Sprintf("Connected to Decred wallet to complete registration at %s, but failed to unlock: %v", dc.acct.host, err)
+						c.notify(newFeePaymentNote("Wallet unlock error", details, db.ErrorLevel, dc.acct.host))
+						result.AuthErr += details
+						return
+					}
+				}
+				c.reFee(dcrWallet, dc)
+				return
+			}
+		}(host, dc)
 	}
 	wg.Wait()
+	c.conns = enabledDEXConnections
```

Comment:
Because this is now writing, the lock at the beginning `c.connMtx.RLock()` needs to have the R (for read) removed for lock and unlock.

Reply:
Thanks for explaining that. Change made for next push.